parser/tools/generate_ast.cc

```diff
@@ -822,7 +822,7 @@ string fieldType(FieldType arg) {
 }
 
 void emitNodeHeader(ostream &out, NodeDef &node) {
-    out << "class " << node.name << " final : public Node {" << '\n';
+    out << "class " << node.name << " : public Node {" << '\n';
```

Comment:
For us to be able to make the `NodeAndExpr` subclasses of these code-genned classes, they can't be `final` anymore. We can undo this once our migration is done.